great_expectations/expectations/regex_based_column_map_expectation.py

```diff
@@ -0,0 +1,266 @@
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+from great_expectations.expectations.metrics.util import get_dialect_regex_expression
+from great_expectations.expectations.util import render_evaluation_parameter_string
+from great_expectations.render.renderer.renderer import renderer
+from great_expectations.render.types import RenderedStringTemplateContent
+from great_expectations.render.util import (
+    handle_strict_min_max,
+    parse_row_condition_string_pandas_engine,
+    substitute_none_for_missing,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ColumnMapRegexMetricProvider(ColumnMapMetricProvider):
+    condition_value_keys = ()
+
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, **kwargs):
+        return column.astype(str).str.contains(cls.regex)
+
+    @column_condition_partial(engine=SqlAlchemyExecutionEngine)
+    def _sqlalchemy(cls, column, _dialect, **kwargs):
+        regex_expression = get_dialect_regex_expression(column, cls.regex, _dialect)
+
+        if regex_expression is None:
+            logger.warning(
+                "Regex is not supported for dialect %s" % str(_dialect.dialect.name)
+            )
+            raise NotImplementedError
+
+        return regex_expression
+
+    @column_condition_partial(engine=SparkDFExecutionEngine)
+    def _spark(cls, column, **kwargs):
+        return column.rlike(cls.regex)
+
+
+class ColumnMapRegexExpectation(ColumnMapExpectation, ABC):
+    @staticmethod
+    def _register_metric(
+        regex_snake_name: str,
+        regex_camel_name: str,
+        regex_: str,
+    ):
+        map_metric = "column_values.match_" + regex_snake_name + "_regex"
+
+        # Define the class using `type`. This allows us to name it dynamically.
+        NewColumnRegexMetricProvider = type(
+            f"(ColumnValuesMatch{regex_camel_name}Regex",
+            (ColumnMapRegexMetricProvider,),
+            {
+                "condition_metric_name": map_metric,
+                "regex": regex_,
+            },
+        )
+
+        return map_metric
+
+    def validate_configuration(self, configuration: Optional[ExpectationConfiguration]):
+        super().validate_configuration(configuration)
+        try:
+            assert (
+                "column" in configuration.kwargs
+            ), "'column' parameter is required for column map expectations"
+            if "mostly" in configuration.kwargs:
+                mostly = configuration.kwargs["mostly"]
+                assert isinstance(
+                    mostly, (int, float)
+                ), "'mostly' parameter must be an integer or float"
+                assert 0 <= mostly <= 1, "'mostly' parameter must be between 0 and 1"
+        except AssertionError as e:
+            raise InvalidExpectationConfigurationError(str(e))
+        return True
+
+    # question, descriptive, prescriptive, diagnostic
```

Comment:
Also, ❤️ for these renderers.

Note: for those that feature "mostly", I'd suggest asking branching logic if Mostly=None or 1, to avoid things like:

Do at least None% of values in column "my_column" match the regular expression `some_regex`?

And 

Do at least 100% of values in column "my_column" match the regular expression `some_regex`?

Instead:

Do all of the values in column "my_column" match the regular expression `some_regex`?

This likely applies to other renderers as well.

Reply:
Actually, thinking about Semantic Typing, I think we can do even better:

`Are at least {mostly * 100}% of values in column "{column}" valid {semantic_type_name_plural}, as judged by matching the regular expression {regex}?`

This would require a new class variable: `semantic_type_name_plural`

Example:

Are at least 80% of values in column "`my_column`" valid Social Security Numbers, as judged by matching the regular expression `\d{3}-\d{3}-\d{2}`?

Example with mostly=1.0 or None:

Are all values in column "`my_column`" valid Social Security Numbers, as judged by matching the regular expression `\d{3}-\d{3}-\d{2}`?

Reply:
My vote would be for `semantic_type_name_plural` to be nullable, with renderers defaulting back to "Do all of the values in column "my_column" match the regular expression some_regex?" if it's None.